### ISSUE TITLE
Get registry to use from config.toml

### ISSUE THREAD

**Issue by qwerty01** [nabijaczleweli/cargo-update#164]
`cargo-update` currently has the index url hard-coded, but sometimes users will have a separate index configured at `~/.cargo/config.toml`

**Comment by qwerty01** [comment]
Oh I misread part of the code, looks like it's not hard-coded, but I'm not sure why it's not getting the right url.

My config.toml looks like this:

```
[source.crates-io]
registry = '[local registry]'
```

and cargo correctly pulls from the local registry, but cargo-update still pulls from github.

**Comment by Logarithmus** [comment]
@qwerty01 the implementation is a bit messed up, looks like the author got custom registries & sources replacements mixed, which are different concepts. I've just started fixing it though, hope to finish it on this weekend.

**Comment by nabijaczleweli** [comment]
`v8.0.0` pulls configuration from `.cargo/config` *and* `.cargo/config.toml` now, so this may well be fixed?

**Comment by Logarithmus** [comment]
@nabijaczleweli Release `v8.0.0` fixed `cargo.toml`, but not the registries. I have a custom private registry (not a mirror of `crates.io`, but containing some in-house developed crates) in my `cargo/config.toml`, like this:
```
[registries]
sample-text = { index = "https://dl.cloudsmith.io/basic/sample-text/private/cargo/index.git" }
```
And when I run `cargo install-update --all` (release `v8.0.0`), I still get this:
```
Couldn't get registry for some-crate: Non-crates.io registry specified and
https://dl.cloudsmith.io/basic/sample-text/private/cargo/index.git couldn't be found in the config file
at .../config.toml. Due to a Cargo limitation we will not be able to install from there until it's given a [source.NAME] in that file!.
```

There are 2 distinct concepts in `cargo`: registries & source replacement. Looking at the source code & an error message, seems you have them mixed up. I don't have source replacements, but I have a custom registry.

**Comment by nabijaczleweli** [comment]
@qwerty01 can you try the current `master` branch (at least af660ba6387949efca93ad5d33a48ba850646857)?

**Comment by nabijaczleweli** [comment]
Absent a response, released in [`v8.1.0`](https://github.com/nabijaczleweli/cargo-update/releases/tag/v8.1.0).
